helm-framework/helm/kubeConfig.go

```diff
@@ -0,0 +1,198 @@
+package helm
+
+import (
+	"log"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/mitchellh/go-homedir"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/discovery/cached/memory"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/restmapper"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+)
+
+// Struct holding k8s client config, burst limit for api requests, and mutex for sync
+type KubeConfig struct {
+	ClientConfig clientcmd.ClientConfig
+	Burst        int
+	sync.Mutex
+}
+
+// Converting KubeConfig to a REST config, which will be used to create k8s clients
+func (k *KubeConfig) ToRESTConfig() (*rest.Config, error) {
+	config, err := k.ClientConfig.ClientConfig()
+	return config, err
+}
+
+// Converting KubeConfig to a discovery client, which will be used to find api resources
+func (k *KubeConfig) ToDiscoveryClient() (discovery.CachedDiscoveryInterface, error) {
+	config, err := k.ToRESTConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	config.Burst = k.Burst
+	return memory.NewMemCacheClient(discovery.NewDiscoveryClientForConfigOrDie(config)), nil
+}
+
+// Converting KubeConfig to a REST mapper, which will be used to map REST resources to their API obj
+func (k *KubeConfig) ToRESTMapper() (meta.RESTMapper, error) {
+	discoveryClient, err := k.ToDiscoveryClient()
+	if err != nil {
+		return nil, err
+	}
+
+	// Use the appropriate types for the arguments
+	mapper := restmapper.NewDeferredDiscoveryRESTMapper(discoveryClient)
+	expander := restmapper.NewShortcutExpander(mapper, discoveryClient)
+
+	return expander, nil
+}
+
+// Function returning raw k8s client config
+func (k *KubeConfig) ToRawKubeConfigLoader() clientcmd.ClientConfig {
+	return k.ClientConfig
+}
+
+// Generates a k8s client config, based on providers settings and namespace, which this config will be used to interact with the k8s cluster
+func (m *Meta) newKubeConfig(namespace *string) (*KubeConfig, error) {
```

Comment:
I'm pretty sure `namespace` doesn't need to be a pointer here – we don't mutate it, not sure why it was this way in the SDKv2 code.